edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/KubernetesExperimentalPodParametersTest.cs

```diff
@@ -338,9 +340,84 @@ public void ParsesSomeDeploymentStrategyExperimentalOptions()
             var parameters = KubernetesExperimentalCreatePodParameters.Parse(experimental).OrDefault();
 
             Assert.True(parameters.DeploymentStrategy.HasValue);
-            var deploymentStrategy = parameters.DeploymentStrategy.OrDefault();
-            Assert.Equal("RollingUpdate", deploymentStrategy.Type);
-            Assert.Equal("1", deploymentStrategy.RollingUpdate.MaxUnavailable);
+            parameters.DeploymentStrategy.ForEach(deploymentStrategy => Assert.Equal("RollingUpdate", deploymentStrategy.Type));
+            parameters.DeploymentStrategy.ForEach(deploymentStrategy => Assert.Equal("1", deploymentStrategy.RollingUpdate.MaxUnavailable));
+        }
+
+        [Fact]
+        public void ParsesEmptyServiceOptions()
+        {
+            var experimental = new Dictionary<string, JToken>
+            {
+                ["k8s-experimental"] = JToken.Parse("{ serviceOptions: {  } }")
+            };
+
+            var parameters = KubernetesExperimentalCreatePodParameters.Parse(experimental).OrDefault();
+
+            Assert.True(parameters.ServiceOptions.HasValue);
+            parameters.ServiceOptions.ForEach(options => Assert.False(options.Type.HasValue));
+            parameters.ServiceOptions.ForEach(options => Assert.False(options.LoadBalancerIP.HasValue));
+        }
+
+        [Fact]
+        public void ParsesServiceOptions()
+        {
+            var experimental = new Dictionary<string, JToken>
+            {
+                ["k8s-experimental"] = JToken.Parse(@"{ serviceOptions: { type: ""nodeport"", loadbalancerip: ""100.1.2.3"" } }")
```

Comment:
loadbalancerip is only valid when servicetype is loadbalancer. Why we honor it when service type is nodeport?

Reply:
Was taking the assumption that we take the data given to us "as is."  There does not seem to be a negative consequence to assigning it even if it is not appropriate, at least in my testing on AKS.

Reply:
I see. What do you see with kubectl describe  the service? Do you see loadbalancerip field when servicetype is nodeport?

Reply:
I have confirmed that a `NodePort` node works like a node port (was able to connect to service through assigned port). The `loadBalancerIP` field is ignored for `NodePort` services. Yes, it shows up in the spec:

```yaml
spec:
  clusterIP: 10.0.150.44
  externalTrafficPolicy: Cluster
  loadBalancerIP: 192.168.1.230
  ports:
  - name: exposedport-6379-tcp
    nodePort: 30570
    port: 6379
    protocol: TCP
    targetPort: 6379
  selector:
    net.azure-devices.edge.deviceid: device1
    net.azure-devices.edge.module: redisedge
  sessionAffinity: None
  type: NodePort
```